agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/JsonConfigPolling.java

```diff
@@ -81,22 +78,23 @@ public void run() {
             FileTime fileTime = attributes.lastModifiedTime();
             if (lastModifiedTime != fileTime.toMillis()) {
                 lastModifiedTime = fileTime.toMillis();
-                Configuration configuration = ConfigurationBuilder.loadJsonConfigFile(path);
+                Configuration config = ConfigurationBuilder.loadJsonConfigFile(path);
+
                 // important to overlay env vars here, so that we don't overwrite the value set by env var
-                ConfigurationBuilder.overlayEnvVars(configuration);
+                ConfigurationBuilder.overlayEnvVars(config);
 
-                if (!configuration.connectionString.equals(TelemetryConfiguration.getActive().getConnectionString())) {
+                if (!config.connectionString.equals(lastReadConfig.connectionString)) {
```

Comment:
i see why you need config object instead of sampling rate here. what is wrong with the old way retrieving the connection string?

Reply:
I don't think any functional difference, but this way the comparison against prior value is the same for across sampling percentage and connection string (and any future fields)